From 628556ba091224f211a7aca7e81ea2e9526626fa Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Mon, 30 May 2016 03:46:36 +0100
Subject: [PATCH 09/13] change_folder_next

Fabian Groffen <grobian@gentoo.org>
* This patch brings back the original behaviour of changer-folder, which
  some people find more useful.
  It suggests the next folder in your mailboxes list that has new mail
  measured from the current folder, instead of the first folder from your
  mailboxes list in any case.  This prevents starvation of folders
  defined at the back of your list ;)
  See discussion: http://thread.gmane.org/gmane.mail.mutt.devel/12457

---
 PATCHES     | 1 +
 curs_main.c | 7 +++++++
 init.h      | 7 +++++++
 mutt.h      | 1 +
 4 files changed, 16 insertions(+)

diff --git a/PATCHES b/PATCHES
index e709cae..2d95814 100644
--- a/PATCHES
+++ b/PATCHES
@@ -1,3 +1,4 @@
+fg.change_folder_next
 vvv.quote
 patch-1.5.4.lpr.collapse_flagged Lukas P. Ruf <lukas.ruf@lpr.ch>
 patch-ifdef-neo-UNKNOWN
diff --git a/curs_main.c b/curs_main.c
index 3021ecc..589e928 100644
--- a/curs_main.c
+++ b/curs_main.c
@@ -1797,7 +1797,14 @@ int mutt_index_menu (void)
 	  }
 	  else
 #endif
+	{
+	  if (option (OPTCHANGEFOLDERNEXT) && Context && Context->path)
+	  {
+	    strfcpy (buf, Context->path, sizeof (buf));
+	    mutt_pretty_mailbox (buf, sizeof(buf));
+	  }
 	  mutt_buffy (buf, sizeof (buf));
+	}
 
           if (mutt_enter_fname (cp, buf, sizeof (buf), &menu->redraw, 1) == -1)
           {
diff --git a/init.h b/init.h
index 723a4e4..23148e7 100644
--- a/init.h
+++ b/init.h
@@ -394,6 +394,13 @@ struct option_t MuttVars[] = {
   ** mailbox by performing a fast mailbox scan when it is defined.
   ** Afterwards the new mail status is tracked by file size changes.
   */
+  { "change_folder_next", DT_BOOL, R_NONE, OPTCHANGEFOLDERNEXT, 0 },
+  /*
+  ** .pp
+  ** When this variable is set, the change-folder command will start at the
+  ** next folder after the current folder in your mailbox list instead of
+  ** starting at the first folder.
+  */
   { "check_new",	DT_BOOL, R_NONE, OPTCHECKNEW, 1 },
   /*
   ** .pp
diff --git a/mutt.h b/mutt.h
index a508539..ee61212 100644
--- a/mutt.h
+++ b/mutt.h
@@ -375,6 +375,7 @@ enum
   OPTBEEP,
   OPTBEEPNEW,
   OPTBOUNCEDELIVERED,
+  OPTCHANGEFOLDERNEXT,
   OPTBRAILLEFRIENDLY,
   OPTCHECKMBOXSIZE,
   OPTCHECKNEW,
-- 
2.8.2

